api/src/main/java/org/openmrs/module/addresshierarchy/config/ConfigLoaderUtil.java

```diff
@@ -0,0 +1,405 @@
+package org.openmrs.module.addresshierarchy.config;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class ConfigLoaderUtil {
+	
+	protected static final String NOT_COMPUTABLE_CHECKSUM = "not_computable_checksum";
+	
+	protected static final String NOT_READABLE_CHECKSUM = "not_readadble_checksum";
+	
+	protected static final String CHECKSUM_FILE_EXT = "checksum";
+	
+	protected static Log log = LogFactory.getLog(ConfigLoaderUtil.class);
+	
+	protected String configDirPath = "";
+	
+	/**
+	 * Non-static instances are based on a configuration subdirectory path.
+	 * 
+	 * @param configDirPath The absolute path to the subdirectory, eg.
+	 *            "../configuration/addresshierarchy"
+	 */
+	public ConfigLoaderUtil(String configDirPath) {
+		if (!StringUtils.isEmpty(configDirPath)) {
+			this.configDirPath = configDirPath;
+		}
```

Comment:
Is there a reason why we want this to be configurable?  Do we not simply want this location to be a static constant located at OpenmrsConstants.APPLICATION_DATA_DIRECTORY + File.separator + "configuration" + File.separator + "addresshierarchy"?